libsolc/libsolc.cpp

```diff
@@ -94,16 +110,42 @@ extern char const* solidity_version() noexcept
 {
 	return VersionString.c_str();
 }
+
 extern char const* solidity_compile(char const* _input, CStyleReadFileCallback _readCallback, void* _readContext) noexcept
 {
-	s_outputBuffer = compile(_input, _readCallback, _readContext);
-	return s_outputBuffer.c_str();
+	return solidity_allocations.emplace_back(compile(_input, _readCallback, _readContext)).data();
 }
-extern void solidity_free() noexcept
+
+extern char* solidity_alloc(size_t _size) noexcept
+{
+	try
+	{
+		solidity_allocations.emplace_back(_size, '\0');
+		return solidity_allocations.back().data();
+	}
+	catch (...)
+	{
+		// most likely a std::bad_alloc(), if at all.
+		return nullptr;
```

Comment:
Is it specified that we can return nullptr?

Reply:
Now it is.